### PR TITLE
CRM-18805 - Contribute - record contribution_page_id on recurring tra…

### PR DESCRIPTION
**Pull request by bsilvern**
…nsactions

---
- CRM-18805: Contribution page not recorded on recurring transactions, Recurring contribution payments do not create CC or BCC emails or profile notifications
  https://issues.civicrm.org/jira/browse/CRM-18805

### CODE REVIEW

**Comment by JoeMurray** [comment]
@eileenmcnaughton here's @bsilvern's updated PR

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -2115,6 +2115,11 @@ protected static function repeatTransaction(&$contribution, &$input, $contributi
       $contributionParams['contact_id'] = $templateContribution['contact_id'];
       $contributionParams['source'] = empty($templateContribution['source']) ? ts('Recurring contribution') : $templateContribution['source'];
 
+      //CRM-18805 -- Contribution page not recorded on recurring transactions, Recurring contribution payments 
+      //do not create CC or BCC emails or profile notifications 
+      if (isset($contribution->contribution_page_id)) //being safe here, not sure if we can ever arrive with this unset
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: can you change the if syntax to a 'full if' with curly braces?

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@seamuslee001 remind me - I think there was some argument AUG were making against this....

**Comment by seamuslee001** [comment]
@eileenmcnaughton I seem to recall Brett didn't like the page information being used for subsequent transactions after the first one as he was feeling the linkage to the contribution page was no longer necessary

**Comment by eileenmcnaughton** [comment]
hmm - I see the relevance in this context though.. do you know where the ticket is?

**Comment by seamuslee001** [comment]
@eileenmcnaughton Redmine 9941 and https://issues.civicrm.org/jira/browse/CRM-18357

**Comment by eileenmcnaughton** [comment]
My current conclusion is we should merge this as it is consistent with previous versions & opt out can be through Alan's new hook

**Comment by seamuslee001** [comment]
Test this please Jenkins

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by JoeMurray** [comment]
@bsilvern Big congrats on getting your first PR merged into CiviCRM core! Great work!

**Pull request merged**